### PR TITLE
✨: Extended AzureRM Credentials

### PR DESCRIPTION
**Pull request by candidson**
## Description

Added ARM_SUBSCRIPTION_ID, ARM_TENANT_ID, ARM_ACCESS_KEY and ARM_ENVIRONMENT as part of the Azure Credentials details

resolves https://github.com/gaia-app/gaia/issues/667

## Checklist for the pull request author

- [x] Tests added for this feature
- [x] Commit messages follow conventions (see [CONTRIBUTING.md](CONTRIBUTING.md))
- [x] Commits are combined ( 1 commit / type and scope )
- [x] Documentation created / updated (if necessary)
- [x] No new sonarqube issues added
- [x] CI pipeline is OK
- [ ] Pull request is fast-forward to the `main` branch
- [ ] The last commit of this merge request :
    - [ ] updates the `pom.xml` to a new minor or major version
    - [ ] updates the `CHANGELOG.md`

## Checklist for the project maintainer

- [ ] Review the code, and add a *Review Approval* if everything is ok
- [ ] This pull request will be merged in merge-commit mode
    - (in github or with the CLI, see [CONTRIBUTING.md](CONTRIBUTING.md))

### After the merge

- [ ] Put the tag on the merged commit
    - (in github or with the CLI, see [CONTRIBUTING.md](CONTRIBUTING.md))
- [ ] Follow the pipeline for the tag, which should build & release the package (maven or docker)

## Issues / Links

* issue : #667

### CODE REVIEW

File: src/test/java/io/gaia_app/credentials/CredentialsTest.kt [hunk 1]
```diff
@@ -14,9 +14,9 @@ internal class CredentialsTest {
     }
 
     @Test
-    fun `toEnv() for AzureRMCredentials should return ARM_CLIENT_ID and ARM_CLIENT_SECRET`() {
-        assertThat(AzureRMCredentials("clientId", "secret").toEnv())
-            .containsExactly("ARM_CLIENT_ID=clientId", "ARM_CLIENT_SECRET=secret")
+    fun `toEnv() for AzureRMCredentials should return ARM_CLIENT_ID, ARM_CLIENT_SECRET, ARM_SUBSCRIPTION_ID, ARM_TENANT_ID, ARM_ACCESS_KEY, ARM_ENVIRONMENT`() {
+        assertThat(AzureRMCredentials("clientId", "secret", "subscriptionId", "tenantId", "backendAccessKey", "environment").toEnv())
```
Comment (juwit) [inline src/test/java/io/gaia_app/credentials/CredentialsTest.kt:18]:
```suggestion
        assertThat(AzureRMCredentials("clientId", "secret", "subscriptionId", "tenantId", "environment", "backendAccessKey").toEnv())
```

**Review by juwit — CHANGES_REQUESTED** [review]
Looks great !
Thanks a lot for this contribution !

The following tests are failing:
io.gaia_app.credentials.CredentialsRestControllerIT#users_shouldBeAbleToCreate_newAzurermCredentials
It's simply missing the new fields you added.

There's also a missing implementation in the VaultCredentialsService:
io.gaia_app.vault.VaultCredentialsServiceKt#encryptAzurermCredentials
and
io.gaia_app.vault.VaultCredentialsServiceKt#decryptAzurermCredentials
Those will correct the failing tests in VaulCredentialsServiceTest

The last failing test should pass after you accept the suggestion.

**Comment by candidson** [comment]
Thanks a lot. I just accepted the changes.

**Comment by juwit** [comment]
:wave: this was merged in #675, so closing this PR

**Pull request closed without merging**